gossip/src/cluster_info.rs

```diff
@@ -664,6 +664,12 @@ impl ClusterInfo {
         self.my_contact_info.read().unwrap().shred_version
     }
 
+    /// Sets the node's own TPU address in gossip cluster info.
+    pub fn set_tpu_address(&self, tpu_address: SocketAddr) {
+        self.my_contact_info.write().unwrap().tpu = tpu_address;
```

Comment:
This change is not enough to change the tpu address. There are threads listening on this socket address, and if you want to change the address then you need to signal to those threads to listen to the new socket address.

But signaling to those threads to listen to a different socket address is going to be a much bigger and intrusive change. I don't think we currently do implement that anywhere else in the code.

So would you please explain what is the context and motivation here?
Maybe there are easier alternatives achieving the same goal.

Reply:
At Blockdaemon, if we're seeing >10 Gbps spikes on our validators, we want to be able to re-route traffic through a load-balanced DDoS scrubbing endpoint to remotely run sigverify and drop spam. The validator currently doesn't offer any such functionality.